src/ripple/app/tx/impl/SetTrust.cpp

```diff
@@ -133,14 +133,22 @@ SetTrust::preclaim(PreclaimContext const& ctx)
     auto const sleDst = ctx.view.read(keylet::account(uDstAccountID));
 
     // If the destination has opted to disallow incoming trustlines
-    // then honour that flag
+    // then honour that flag ONLY if the trustline doesn't exist
     if (ctx.view.rules().enabled(featureDisallowIncoming))
     {
         if (!sleDst)
             return tecNO_DST;
 
         if (sleDst->getFlags() & lsfDisallowIncomingTrustline)
-            return tecNO_PERMISSION;
+        {
+            if (ctx.view.rules().enabled(fixDisallowIncomingV1) &&
+                ctx.view.exists(keylet::line(id, uDstAccountID, currency)))
+            {
+                // pass
+            }
+            else
+                return tecNO_PERMISSION;
```

Comment:
Why not just invert the `if` and return inside it, instead of having an empty `if` and returning in the `else`?

Reply:
I actually have it like that on my local branch but with `clang-format` it makes it very "compact". I'm open to doing it like that if its preferred. It was just done like that for readability.

```
if (ctx.view.rules().enabled(fixDisallowIncomingV1) &&
        !ctx.view.exists(keylet::line(id, uDstAccountID, currency)))
        return tecNO_PERMISSION;
```

Reply:
In general positive conditions should be preferred over negative conditions. They are quick and easy to scan through, it's very obvious to the reader what's happening. The more complicated the condition the more likely the next programmer will misunderstand it and insert a mistake.